internal/api/lib.go

```diff
@@ -52,11 +74,15 @@ func InitCache(dataDir string, supportedCapabilities []string, cacheSize uint32,
 	if err != nil {
 		return Cache{}, errorWithMessage(err, errmsg)
 	}
-	return Cache{ptr: ptr}, nil
+	return Cache{ptr: ptr, lockfile: *lockfile}, nil
 }
 
 func ReleaseCache(cache Cache) {
 	C.release_cache(cache.ptr)
+
+	// Release directory lock
+	_ = syscall.Flock(int(cache.lockfile.Fd()), syscall.LOCK_UN)
+	cache.lockfile.Close()
```

Comment:
Closing the file releases the lock anyways, so no need for the explicit syscall

Reply:
Oh, are you sure? I found it this way in some example. Don't know. Should we remove the unlock?

Reply:
The [linux manpage](https://linux.die.net/man/2/flock) for `flock` mentions:
> [...] the lock is released either by an explicit LOCK_UN operation on any of these duplicate descriptors, or when all such descriptors have been closed. 

For [bsd](https://man.freebsd.org/cgi/man.cgi?query=close&sektion=2&apropos=0&manpath=FreeBSD+14.0-RELEASE+and+Ports), it's described in the page about `close`:
> on the last close of a file holding an advisory lock the lock is released

Whether to remove it or not is probably more of a taste question, since this is not a hot path (In fact, as far as I can see, wasmd never calls this).

Reply:
and on my Macbook, the `TestInitLockingPreventsConcurrentAccess` test still passes if I remove the syscall

Reply:
I'll remove it and just add a comment that closing the file releases the lock for simplicity